### PR TITLE
[feature] custom SVG cursors

### PR DESCRIPTION
**Pull request by steveruizok**
This PR begins the work of implementing custom SVG cursors for the app.

The `CursorManager` class is used to update the `cursor` style of the tldraw component wrapper using SVG strings. We'll need to work out how (or whether) to override the cursor settings coming from the tldraw/core renderer components, such as the bounds handles.

### CODE REVIEW

**Comment by Soyn** [comment]
After change the tool, the default cursor is still grab. Is it right?

File: packages/tldraw/src/state/CursorManager.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    if (!elm) return
+
+    const cursor = this.current
+      ? `url("${CursorManager.cursorDataUris[this.current]}") 17.5 17.5, auto`
+      : `auto`
+
+    elm.style.setProperty('cursor', cursor)
+  }
+
+  static cursorDataUris: Record<Cursor, string> = {
```
Comment (cagataycali) [inline packages/tldraw/src/state/CursorManager.ts:75]: Can we serve these URL's from a CDN for a better bundle size?

**Comment by Francois-Laberge-Bose** [comment]
I came to say the same thing as @Soyn. Here's a video demonstrating the issue.

**Repro steps:**
 1. Go to https://tldraw-git-feature-svg-cursors-steveruiz.vercel.app/
 2. Activate the Drawing tool and create some strokes
 3. Activate the Selection tool, select the strokes you just created, and grab the rotation handle
 4. Deselect the shape
 5. Notice that the cursor stays as the grab cursor when
   6. Drawing out a selection rect
   7. When drawing an arrow
   8. I'm sure more things

https://user-images.githubusercontent.com/34040155/148782767-d3c4e85b-3dd5-4330-ba25-ae95578359b8.mov

**Comment by steveruizok** [comment]
Closing this as I think I have a better implementation that I can backport from the v2.

**Pull request closed without merging**